File: backend/LoanManagement.Entities/Models/Customer.cs
namespace LoanManagement.Entities.Models;

public class Customer
{
    public int Id { get; set; }
    public string FirstName { get; set; } = null!;
    public string LastName { get; set; } = null!;
    public string IdentityNumber { get; set; } = null!;
    public string Email { get; set; } = null!;
    public string PhoneNumber { get; set; } = null!;
    public bool IsDeleted { get; set; }
    public DateTime? DeletedAtUtc { get; set; }
    public virtual ICollection<Loan>? Loans { get; set; } = new List<Loan>(); //müşteriye ait kredi kayıtları (Navigation Property)
}

File: frontend/src/types/customer.ts
import type { LoanSummaryDto, LoanDetailSummaryDto } from "./loan";

export interface CreateCustomerDto {
  firstName: string;
  lastName: string;
  identityNumber: string;
  email: string;
  phoneNumber?: string | null;
}

export interface UpdateCustomerDto {
  email: string;
  phoneNumber?: string | null;
}

export interface CustomerListDto {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string | null;
  totalLoans: number;
  loans: LoanSummaryDto[];
  /**
   * Müşteri soft-delete edilmişse true. Admin tablosunda "Aktif/Pasif" badge'i
   * ve dinamik aksiyon butonu (Sil ↔ Aktife Çevir) için kullanılır.
   * Backend: `LoanManagement.Entities.DTOs.CustomerListDto.IsDeleted` (v1.4+).
   */
  isDeleted: boolean;
  /** Soft-delete zamanı (UTC, ISO-8601). `isDeleted=false` ise `null`. */
  deletedAtUtc: string | null;
}

export interface CustomerResponseDto extends CustomerListDto {
  identityNumber: string;
}

export interface CustomerSummaryDto {
  customerId: number;
  fullName: string;
  totalLoans: number;
  activeLoans: number;
  closedLoans: number;
  totalDebt: number;
  totalPaid: number;
  totalInstallments: number;
  paidInstallments: number;
  unpaidInstallments: number;
  overdueInstallments: number;
  loans: LoanDetailSummaryDto[];
}

export interface CreateCustomerResponse {
  id: number;
  message: string;
}

export interface MessageResponse {
  message: string;
}
